R/drop_upload.R

```diff
@@ -23,37 +27,59 @@
 #' drop_upload("mtt.csv")
 #'}
 drop_upload <- function(file,
-                        dest = NULL,
-                        overwrite = TRUE,
-                        autorename = FALSE,
+                        path = NULL,
+                        mode = "overwrite",
```

Comment:
We could match the arg order exactly, but I'd rather our package serve as a functional interface to the API. My rationale is that if people are using this package for data science tasks, then they are writing outputs from scripts to Dropbox. So running the same analysis dozens of times should just overwrite the output files, and not create a proliferation of `output.csv` `output (1).csv` `output (2).csv` ... `output(400).csv` and so on. That would be maddening. 

What do you think?

Reply:
yeah, that's a good point, and seeing filenames written out like that remind me of when I didn't know git...the dark times.

I wonder if this is where `mode = "update"` would help; maybe that allows us to add a new revision rather than a revision-destroying full-overwrite? in any case, you've convinced me this is the right behaviour in the near term

Reply:
👍 I will add an issue to investigate the update mode (even for this CRAN submission). For now my priority is to get this PR merged. I think there is a use case for update. It will require a working `drop_dir` since I'll need that to check the `rev`.

Reply:
I've left the default mode to "overwrite" but will change it to "add" if you feel strongly.